reddwarf/types/agora.py

```diff
@@ -0,0 +1,44 @@
+from typing import TypedDict, List, Optional
+from enum import IntEnum
+
+IncrementingId = int
+
+class Statement(TypedDict):
+     id: IncrementingId
+
+class Participant(TypedDict):
+    id: IncrementingId
+
+class ClusteredParticipant(TypedDict):
+    id: IncrementingId # participant.id
+    x: float
+    y: float
+
+class VoteValueEnum(IntEnum):
+    AGREE = 1
+    DISAGREE = -1
+    # Can withhold using "pass" at own discretion.
+    PASS = 0
+
+class Vote(TypedDict):
+    statement_id: IncrementingId # statement.id
+    participant_id: IncrementingId # participant.id
+
+    vote: VoteValueEnum
+
+class Conversation(TypedDict):
+    votes: List[Vote]
```

Comment:
```suggestion
class Conversation(TypedDict):
    votes: List[Vote]
```

The minimal interface requires only a list of votes, with `participant_id` (row id), `statement_id` (column id) and `vote` value.

The main reason why an optional list of statements would be useful is if we want that list to be an implicit filter. aka, only votes from statements id's from that set are assumed part of the calculation, when included. The same could be true of participants.

```
class Conversation(TypedDict):
    votes: List[Vote]
    statements: Optional[List[Statement]]
    participants: Optional[List[Participants]]
```

Would this be helpful? It would essentially be treating `statements` and `participants` keys as optional "include this entity" args, where if either is absent, it's assumed all are included.

Reply:
Obviously, this filtering of votes by participant_id and statement_id can be vote in advance on the votes record list, so this isn't required, and maybe strays beyond minimal.

Reply:
I think just the `votes` one seems sufficient!